confmap/provider/eecprovider/watcher.go

```diff
@@ -0,0 +1,51 @@
+// Copyright Dynatrace LLC
+// SPDX-License-Identifier: Apache-2.0
+
+package eecprovider // import "github.com/Dynatrace/dynatrace-otel-collector/confmap/provider/eecprovider"
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/collector/confmap"
+)
+
+type watcher struct {
+	providerCtx context.Context
+	reqCtx      context.Context
+
+	getConfigBytes func() ([]byte, error)
+
+	refreshInterval time.Duration
+	watcherFunc     confmap.WatcherFunc
+	configHash      [32]byte
+}
+
+func (w *watcher) watchForChanges() {
+	ticker := time.NewTicker(w.refreshInterval)
+
+	for {
+		select {
+		case <-ticker.C:
+			body, err := w.getConfigBytes()
```

Comment:
What happens if this takes longer than `refreshInterval`? Should we terminate the HTTP request if that happens?

Reply:
That's a really good observation, I overlooked that. I've updated it so we're properly propagating context down to the request.